process/transactionLog/process.go

```diff
@@ -84,34 +86,34 @@ func (tlp *txLogProcessor) GetLog(txHash []byte) (data.LogHandler, error) {
 
 // GetLogFromCache retrieves a log generated by a transaction from ram or from storage
 // TODO remove this function and change the interface no longer needed
-func (tlp *txLogProcessor) GetLogFromCache(txHash []byte) (data.LogHandler, bool) {
+func (tlp *txLogProcessor) GetLogFromCache(txHash []byte) (*data.LogData, bool) {
 	tlp.mut.RLock()
 	defer tlp.mut.RUnlock()
 
-	txLogFromCache, ok := tlp.logs[string(txHash)]
+	index, ok := tlp.logsIndices[string(txHash)]
 	if ok {
-		return txLogFromCache, true
+		return tlp.logs[index], true
 	}
 
 	txLog, err := tlp.GetLog(txHash)
 	if err != nil {
 		return nil, false
 	}
 
-	return txLog, true
+	return &data.LogData{
+		LogHandler: txLog,
+		TxHash:     string(txHash),
+	}, true
 }
 
 // GetAllCurrentLogs will return all generated logs for the current block
-func (tlp *txLogProcessor) GetAllCurrentLogs() map[string]data.LogHandler {
-	logsMap := make(map[string]data.LogHandler)
-
+func (tlp *txLogProcessor) GetAllCurrentLogs() []*data.LogData {
 	tlp.mut.RLock()
-	for key, value := range tlp.logs {
-		logsMap[key] = value
-	}
+	logsSlice := make([]*data.LogData, 0, len(tlp.logs))
+	logsSlice = append(logsSlice, tlp.logs...)
```

Comment:
is there a particular reason to not just set `logsSlice := tlp.logs` ?

Reply:
yes, it's safer to make a clone of the slice